packages/coil-extension/src/background/services/BackgroundScript.ts

```diff
@@ -725,14 +725,28 @@ export class BackgroundScript {
 
     const receiver = stream.getPaymentPointer()
     const { assetCode, assetScale, exchangeRate } = stream.getAssetDetails()
+
+    // Set tip amount
     const tipAmount = tip < 1 ? 1 : tip
     const amount = Math.floor(tipAmount * 1e9 * exchangeRate).toString() // 1 USD, assetScale = 9
 
+    // Set tip origin url
+    async function tabQueryPromise() {
+      return new Promise(function (res, rej) {
+        chrome.tabs.query({ active: true, currentWindow: true }, res)
+      })
+    }
+    const activeTabUrl = await tabQueryPromise().then(
+      (tabs: any) => tabs[0].url
```

Comment:
This requires tabs permission on firefox (which we may end up using, but still .. )
We do track the current url per frame using webNavigation apis
It would be best if you got it from the frames service

Reply:
Pretty it, add convenience functions however you like, but something like this:

```typescript
  getActiveTabUrl() {
    const tabId = this.activeTab
    // Get the url for the top frame
    const frameId = 0
    const frame = 
      notNullOrUndef(this.framesService.getFrame({frameId, tabId}))
    return frame.href
  }
```